cmd/allocator/main.go

```diff
@@ -323,6 +324,7 @@ func (h *httpHandler) allocateHandler(w http.ResponseWriter, r *http.Request) {
 
 func httpCode(err error) int {
 	code := http.StatusInternalServerError
+	// nolint: gocritic
```

Comment:
Does it want to turn this into an if statement? A switch with one case seems strange.

Reply:
Yes, but it is a type switch, so it seems there is no way to switch this structure into an if statement.
 https://github.com/golang/go/wiki/Switch#type-switch

Reply:
The only way could be this syntax:
```
if t, ok := err.(k8serror.APIStatus) ; ok {
 ...
}
```

Reply:
Both options will be equivalent:
https://play.golang.org/p/LkcbMXHC6gU

Reply:
By the way, same one case switch could be found here:
https://github.com/kubernetes/apimachinery/blob/51168acb8295af7876446c9e6a5e77aee0a838a7/pkg/api/errors/errors.go#L577

Reply:
Clearly they aren't running gocritic. :)

I'm guessing that the tool prefers that you use the equivalent syntax (explicitly checking for the desired type) makes sense when you are checking for a single type, because then you have a simple if statement instead of a switch statement with a single case.